nsxt/resource_nsxt_lb_http_forwarding_rule_test.go

```diff
@@ -0,0 +1,289 @@
+		if !ok {
+			return fmt.Errorf("NSX LB rule resource %s not found in resources", resourceName)
+		}
+
+		resourceID := rs.Primary.ID
+		if resourceID == "" {
+			return fmt.Errorf("NSX LB rule resource ID not set in resources ")
+		}
+
+		monitor, responseCode, err := nsxClient.ServicesApi.ReadLoadBalancerRule(nsxClient.Context, resourceID)
+		if err != nil {
+			return fmt.Errorf("Error while retrieving LB rule with ID %s. Error: %v", resourceID, err)
+		}
+
+		if responseCode.StatusCode != http.StatusOK {
+			return fmt.Errorf("Error while checking if LB rule %s exists. HTTP return code was %d", resourceID, responseCode.StatusCode)
+		}
+
+		if displayName == monitor.DisplayName {
+			return nil
+		}
+		return fmt.Errorf("NSX LB rule %s wasn't found", displayName)
+	}
+}
+
+func testAccNSXLbHTTPForwardingRuleCheckDestroy(state *terraform.State, displayName string) error {
+	nsxClient := testAccProvider.Meta().(*nsxt.APIClient)
+	for _, rs := range state.RootModule().Resources {
+
+		if rs.Type != "nsxt_lb_http_forwarding_rule" {
+			continue
+		}
+
+		resourceID := rs.Primary.Attributes["id"]
+		monitor, responseCode, err := nsxClient.ServicesApi.ReadLoadBalancerRule(nsxClient.Context, resourceID)
+		if err != nil {
+			if responseCode.StatusCode != http.StatusOK {
+				return nil
+			}
+			return fmt.Errorf("Error while retrieving LB rule with ID %s. Error: %v", resourceID, err)
+		}
+
+		if displayName == monitor.DisplayName {
+			return fmt.Errorf("NSX LB rule %s still exists", displayName)
+		}
+	}
+	return nil
+}
+
+func testAccNSXLbHTTPForwardingRuleCreateTemplate(matchStrategy string, matchType string, caseSensitive string, inverse string) string {
+	return fmt.Sprintf(`
+resource "nsxt_lb_pool" "pool" {
+  description = "test description"
+}
+
+resource "nsxt_lb_http_forwarding_rule" "test" {
+  display_name   = "test"
+  description    = "test description"
+  match_strategy = "%s"
+`, matchStrategy) +
+		testAccNSXLbRuleValueConditionTemplate("body", "value", "EQUALS", "false", "false", 1) +
+		testAccNSXLbRuleValueConditionTemplate("body", "value", matchType, caseSensitive, inverse, 2) +
+		testAccNSXLbRuleNameValueConditionTemplate("header", "EQUALS", "false", "false", 1) +
+		testAccNSXLbRuleNameValueConditionTemplate("header", matchType, caseSensitive, inverse, 2) +
+		testAccNSXLbRuleNameValueConditionTemplate("cookie", matchType, caseSensitive, inverse, 1) +
+		testAccNSXLbRuleSimpleConditionTemplate("method", "HEAD", inverse) +
+		testAccNSXLbRuleSimpleConditionTemplate("version", "HTTP_VERSION_1_1", inverse) +
+		testAccNSXLbRuleSimpleMatchConditionTemplate("uri", "/hello", matchType, inverse) +
+		fmt.Sprintf(`
+  ip_condition {
+    source_address = "1.1.1.1"
+    inverse        = %s
+  }
+
+  tcp_condition {
+    source_port = "7887"
+    inverse     = %s
+  }
+
+  http_reject_action {
+    reply_status  = "500"
+    reply_message = "rejected"
+  }
+
+  http_redirect_action {
```

Comment:
its weird that NSX allows to chain these together..

Reply:
Yeh...